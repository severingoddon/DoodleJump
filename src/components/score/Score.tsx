import React from 'react';
import './score-css.css';

let scoreClass = 'score-field';

export const setThemeScore = (isDark: boolean) =>{
    if(isDark) scoreClass = 'score-field-dark';
    else scoreClass = 'score-field';
};

export default (props: any) => {
    return(
        <div data-testid="score" className={scoreClass}>
            {props.scoreText}
        </div>
    );
};