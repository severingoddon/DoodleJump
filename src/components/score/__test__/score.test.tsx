import React from 'react';
import ReactDOM from 'react-dom';
import Score from '../Score';
import TestRenderer from 'react-test-renderer';
import { render, cleanup  } from '@testing-library/react';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Score scoreText="0"/>, div);
});

it('renders score correctly', () => {
    const {getByTestId} = render(<Score scoreText="55"/>);
    expect(getByTestId('score')).toHaveTextContent('55');
});

it('matches snapshot', () => {
    const tree = TestRenderer.create(<Score scoreText={'55'}/>).toJSON();
    expect(tree).toMatchSnapshot();
});



