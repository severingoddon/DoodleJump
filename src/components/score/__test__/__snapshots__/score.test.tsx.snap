// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="score-field"
  data-testid="score"
>
  55
</div>
`;
